nucliadb_sdk/src/nucliadb_sdk/v2/sdk.py

```diff
@@ -95,31 +99,315 @@
+    ),
+    "summarize": SDKDefinition(
+        path_template="/v1/kb/{kbid}/summarize",
+        method="POST",
+        path_params=("kbid",),
+    ),
+    "feedback": SDKDefinition(
+        path_template="/v1/kb/{kbid}/feedback",
+        method="POST",
+        path_params=("kbid",),
+    ),
+    "start_export": SDKDefinition(
+        path_template="/v1/kb/{kbid}/export",
+        method="POST",
+        path_params=("kbid",),
+    ),
+    "export_status": SDKDefinition(
+        path_template="/v1/kb/{kbid}/export/{export_id}/status",
+        method="GET",
+        path_params=("kbid", "export_id"),
+    ),
+    "download_export": SDKDefinition(
+        path_template="/v1/kb/{kbid}/export/{export_id}",
+        method="GET",
+        path_params=("kbid", "export_id"),
+    ),
+    "create_kb_from_import": SDKDefinition(
+        path_template="/v1/kbs/import",
+        method="POST",
+        path_params=(),
+    ),
+    "start_import": SDKDefinition(
+        path_template="/v1/kb/{kbid}/import",
+        method="POST",
+        path_params=("kbid",),
+    ),
+    "import_status": SDKDefinition(
+        path_template="/v1/kb/{kbid}/import/{import_id}/status",
+        method="GET",
+        path_params=("kbid", "import_id"),
+    ),
+    "trainset": SDKDefinition(
+        path_template="/v1/kb/{kbid}/trainset",
+        method="GET",
+        path_params=("kbid",),
+    ),
+    # Learning Configuration
+    "get_configuration": SDKDefinition(
+        path_template="/v1/kb/{kbid}/configuration",
+        method="GET",
+        path_params=("kbid",),
+    ),
+    "set_configuration": SDKDefinition(
+        path_template="/v1/kb/{kbid}/configuration",
+        method="POST",
+        path_params=("kbid",),
+    ),
+    # Learning models
+    "download_model": SDKDefinition(
+        path_template="/v1/kb/{kbid}/models/{model_id}/{filename}",
+        method="GET",
+        path_params=("kbid", "model_id", "filename"),
+    ),
+    "get_models": SDKDefinition(
+        path_template="/v1/kb/{kbid}/models",
+        method="GET",
+        path_params=("kbid",),
+    ),
+    "get_model": SDKDefinition(
+        path_template="/v1/kb/{kbid}/model/{model_id}",
+        method="GET",
+        path_params=("kbid", "model_id"),
+    ),
+    # Learning config schema
+    "get_configuration_schema": SDKDefinition(
+        path_template="/v1/kb/{kbid}/schema",
+        method="GET",
+        path_params=("kbid",),
+    ),
+    # Custom synonyms
+    "set_custom_synonyms": SDKDefinition(
+        path_template="/v1/kb/{kbid}/custom-synonyms",
+        method="PUT",
+        path_params=("kbid",),
+    ),
+    "get_custom_synonyms": SDKDefinition(
+        path_template="/v1/kb/{kbid}/custom-synonyms",
+        method="GET",
+        path_params=("kbid",),
+    ),
+}
 
 
-def ask_response_parser(response: httpx.Response) -> SyncAskResponse:
+def ask_response_parser(response_type: Type[BaseModel], response: httpx.Response) -> BaseModel:
```

Comment:
why do you need to pass `response_type` as a parameter here? isn't it always `SyncAskResponse`?
I think the return type of this function should be `SyncAskResponse`.

And also, for the async version, I think we shuld be iterating the response asynchronously no? (that wasnt implemented before either)

Reply:
I'm not changing the API SDK behavior on this PR, just fixing the typing. Can not be SyncAskResponse because the Generic is bound to BaseModel.